app/src/main/java/org/simple/clinic/navigation/v2/NavRequest.kt

```diff
@@ -0,0 +1,15 @@
+package org.simple.clinic.navigation.v2
+
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
+sealed class NavRequest(val key: ScreenKey) : Parcelable
+
+@Parcelize
+data class Normal(private val _key: ScreenKey) : NavRequest(_key)
+
+@Parcelize
+data class ExpectingResult(
+    val requestType: Parcelable,
```

Comment:
This is a replacement for `startActivityForResult(requestCode, intent)`. Instead of using integer request codes to identify why an activity was opening, I decided to make it a `Parcelable`. This will enforce stronger typing on the request codes and reduce chances of errors by using hardcoded integer request codes, or reusing request code values for the same activity from different locations.